### PR TITLE
For #18608: Also add default browser item in new menu.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mcarare** [comment]
With the new menu also in Nightly, in order to run the experiment on Nightly the item should also be added to this menu, not only in the old menu that only exists in Beta and Release now.
cc @Amejia481 for confirmation.

**Review by Amejia481 — APPROVED** [review]
Thanks for this patch!

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -561,6 +563,8 @@ open class DefaultToolbarMenu(
                 extensionsItem,
                 if (tabsTrayRewrite) syncMenuItem else syncedTabsItem,
                 BrowserMenuDivider(),
+                getSetDefaultBrowserItem(),
+                BrowserMenuDivider(),
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:567]: If the item is not shown the divider will show either way.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 2]
```diff
@@ -362,7 +362,9 @@ open class DefaultToolbarMenu(
             BrowserMenuDivider(),
             reportSiteIssuePlaceholder,
             findInPage,
+            BrowserMenuDivider(),
             getSetDefaultBrowserItem(),
+            BrowserMenuDivider(),
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:367]: If the item is not shown the divider will show either way.

**Review by Amejia481 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 2 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:367]: As `getSetDefaultBrowserItem()` returns null when the experiment is not active

Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: Good catch. I am showing them depending on the value now. Not sure, but I think one of them should be shown either way.

**Review by Amejia481 — APPROVED** [review]
🚀 !

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 2 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: 👍🏽

**Comment by Amejia481** [comment]
@Mergifyio backport releases_v89.0.0

**Pull request merged**